### PR TITLE
[IR] Make isHidden and isAssignable explicit on IrValueParameter.

### PR DESCRIPTION
**Pull request by madsager**
There were a couple of places where they were confused and
isAssignable was passed as a positional parameter in the position
of isHidden.

### CODE REVIEW

**Comment by madsager** [comment]
kotlin/native change: https://github.com/JetBrains/kotlin-native/pull/4558

File: compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt [hunk 1]
```diff
@@ -351,6 +351,7 @@ open class DeepCopyIrTreeWithSymbols(
             declaration.varargElementType?.remapType(),
             declaration.isCrossinline,
             declaration.isNoinline,
+            declaration.isHidden,
```
Comment (madsager) [inline compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt:354]: This was the bug. declaration.isAssignable was passed in the isHidden position.

**Comment by madsager** [comment]
Friendly ping. :)

**Review by romanart — APPROVED** [review]
LGTM

**Comment by madsager** [comment]
This fixes the default argument handling for compose. Therefore, it would be great to get this in and hopefully cherry-picked to 1.4.30. We need to make sure to get the native part merged as well.

**Comment by udalov** [comment]
Looks good, thanks!

**Comment by udalov** [comment]
Merged to [master](https://github.com/JetBrains/kotlin/commit/5d9e86863a33c9c6969fb2747b4edbc00d8697bc) and [1.4.30](https://github.com/JetBrains/kotlin/commit/f9ead560215bf4af94541c62f841ff9c6ea103f4)

**Pull request merged**